src/components/map/layers/new/NewRoutePathLayer.tsx

```diff
@@ -112,6 +112,23 @@ export default class RoutePathLayer extends Component<IRoutePathLayerProps> {
         return this.renderNode(firstNode, link, { isNeighbor: false }, 0);
     }
 
+    async componentDidUpdate() {
```

Comment:
Why do we have this in componentDidUpdate, instead of just calling it as a function where it is needed? From my limited understanding of this chain of events, would it be possible to call it after addLinkToRoutePath?